masonry/src/passes/update.rs

```diff
@@ -319,6 +322,50 @@ pub(crate) fn run_update_disabled_pass(root: &mut RenderRoot) {
 
 // ----------------
 
+#[allow(clippy::only_used_in_recursion)]
+fn update_stashed_for_widget(
+    global_state: &mut RenderRootState,
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    state: ArenaMut<'_, WidgetState>,
+    parent_stashed: bool,
+) {
+    let _span = widget.item.make_trace_span().entered();
+    let id = state.item.id;
+
+    let stashed = state.item.is_explicitly_stashed || parent_stashed;
+    if !state.item.needs_update_stashed && stashed == state.item.is_stashed {
+        return;
+    }
+
+    if stashed != state.item.is_stashed {
+        // TODO - Send update event
+        state.item.is_stashed = stashed;
+        state.item.update_focus_chain = true;
```

Comment:
Does this need to also request a repaint?

Reply:
Good catch.